src/util/native_fs_utils.js

```diff
@@ -548,6 +548,9 @@ async function is_dir_rw_accessible(fs_context, dir_path) {
     } catch (err) {
         return false;
     }
+
+    if (config.NC_DISABLE_POSIX_MODE_ACCESS_CHECK) return true;
```

Comment:
Can we move this check at the start of the method? for checking path validity we have another method is_path_exists()

Reply:
no, this is here because we still want to do an access check just not do mode bits check because it's not supporting ACLs.